### PR TITLE
Make HNC easily deployable onto non-Kind clusters

### PR DESCRIPTION
**Pull request by adrianludwin**
/assign @rjbez17 
/assign @yiqigao217 

Also fixes a bug that `deploy` doesn't delete the old deployment first.

### CODE REVIEW

File: incubator/hnc/Makefile [hunk 1]
```diff
@@ -91,22 +76,50 @@ else
...
-# Creates a local kind cluster, destroying the old one if necessary. It's not
-# *necessary* to call this wih CONFIG=kind but it's not a bad idea either so
-# the correct manifests get created.
-kind-reset: manifests
-	@echo "Warning: you must have sourced 'devenv' so kubectl points to the Kind cluster"
+###################### DEPLOYABLE ARTIFACTS AND ACTIONS #########################
+
+# Deploy controller in the configured Kubernetes cluster in ~/.kube/config
+deploy: deploy-prereq docker-push kubectl manifests
+	-kubectl -n hnc-system delete deployment hnc-controller-manager
```
Comment (rjbez17) [inline incubator/hnc/Makefile]: Any reason not to use `kubectl delete -f manifests/hnc-manager.yaml` here?

Comment (adrianludwin) [reply incubator/hnc/Makefile]: Good call, fixed.

Comment (adrianludwin) [reply incubator/hnc/Makefile]: Actually, this deletes the CRDs so any existing hierarchy configs are wiped out too. Oops.

Comment (adrianludwin) [reply incubator/hnc/Makefile]: I'm reverting this back to the way it was.

Comment (adrianludwin) [reply incubator/hnc/Makefile]: I updated the comment now, lgty?

Comment (rjbez17) [reply incubator/hnc/Makefile]: 👍

**Pull request merged**